lib/pdk/module/templatedir.rb

```diff
@@ -221,6 +223,7 @@ def config_for(dest_path)
           if File.file?(config_path) && File.readable?(config_path)
             begin
               @config = YAML.safe_load(File.read(config_path), [], [], true)
+              @config.merge!('global' => { 'userdata' => @userdata })
```

Comment:
Rather than merging this into the `@config` hash (which i'm planning to eventually deprecate), it would be cleaner (imho) to add this as the `module_metadata` key in the `file_config` hash, and then access it in the templates as `module_metadata['version']` etc.